### ISSUE TITLE
"No Lock" with Hauppauge WinTV-quadHD and firmware 6.0.13

### ISSUE THREAD

**Issue by citronalco** [MythTV/mythtv#772]
* **Platform**:
Debian Linux 12

* **MythTV version**:
2:33.1+fixes.20230704.83e1df82c2

* **Package version**:

* **Component**:
mythtv-backend

### What steps will reproduce the bug?

Hauppauge WinTV-quadHD (Europe) depends on firmware file dvb-demod-si2168-d60-01.fw.
If this file is version 6.0.13 as currently shipped with Debian 12, MythTV does not get a channel lock, neither in channel scan nor while live tv or recording. Other programs like TV Headend work properly.

Downgrading the firmware e.g. to 6.0.1 fixes this MythTV issue.

### How often does it reproduce? Is there a required condition?
always
### What do you see instead?

"TuningSignalCheck: SignalMonitor timed out." in Myth-Backend's log, 0 byte recordings in MythWeb

**Comment by Jpilk** [comment]
I'm not familiar with Debian and have found no references to the 6.0.13 firmware. 

In Fedora  with master , and different dvb hardware,  I  see 

[john@HPFed ~]$ dmesg | grep -i dvb-demod
[  368.246156] si2168 1-0064: downloading firmware from file 'dvb-demod-si2168-b40-01.fw'
[john@HPFed ~]$

Just one load soon(ish) after boot.  IIRC that file came from 

https://github.com/OpenELEC/dvb-firmware/blob/master/firmware/dvb-demod-si2168-b40-01.fw

and is retrieved from  /lib/firmware  

Do you have two different copies?

**Comment by kmdewaal** [comment]
There have been issues with the DVB drivers in various 6.x versions of the Linux kernel.
Once you have a fully stable situation again with an old kernel and the 6.0.1 firmware you could try replacing only the firmware file with the 6.0.13 version and see what happens.
This would then indicate whether it is the kernel version or the firmware version that makes the difference.
If it is the kernel version then it will sort itself out eventually once the fixes for the DVB drivers propagate into the kernels in the distributions.

**Comment by Jpilk** [comment]
Like Klaas, I wondered about the kernel version.  I had problems in Fedora from 6.3.6, ok now with 6.3.12

The sizes of the downloads from your two URLs above are very different. 

[john@HP_Box Downloads]$ ls -ls dvb*
12 -rw-rw-r--. 1 john john 10829 Jul 18 17:04 dvb-demod-si2168-d60-01armbian.fw
 4 -rw-rw-r--. 1 john john   204 Jul 18 17:13 dvb-demod-si2168-d60-01palos.fw
[john@HP_Box Downloads]$

**Comment by Jpilk** [comment]
The issue #761 that I saw in Fedora is no longer causing me obvious harm; but there was also an amd-related problem that has apparently been fixed in the latest kernel. I don't know if it might be relevant here, or how long it will take to reach the repos. 

https://gitlab.freedesktop.org/drm/amd/-/issues/2658#note_2004564

**Comment by citronalco** [comment]
All versions of dvb-demod-si2168-b40-01.fw have different sizes. There's no padding. The higher the version, the bigger the size.

**Comment by citronalco** [comment]
No, happens also with the older Linux 6.1.0-0.deb11.7-amd64 kernel. MythTV does not work, TVHeadend does.
